commcare_connect/opportunity/api/serializers.py

```diff
@@ -98,3 +99,21 @@ class UserVisitSerializer(serializers.ModelSerializer):
     class Meta:
         model = UserVisit
         fields = ["id", "status", "visit_date", "deliver_form_name", "deliver_form_xmlns"]
+
+
+class PaymentSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Payment
+        fields = ["amount", "date_paid"]
+
+
+class DeliveryProgressSerializer(serializers.Serializer):
+    deliveries = serializers.SerializerMethodField()
+    payments = serializers.SerializerMethodField()
+
+    def get_payments(self, obj):
+        return [PaymentSerializer(payment) for payment in obj.payment_set.all()]
```

Comment:
This serializer can be changed to use `many=True` that will automatically serialize all the objects in the queryset to a ListSerializer. 
```suggestion
        return PaymentSerializer(obj.payment_set.all(), many=True)
```

Reply:
nice, I was trying to figure out how to use `many` in the attribute level fields, but hadn't considered I could use it in the methods